### ISSUE TITLE
Tagged release missing submodules files

### ISSUE THREAD

**Issue by mxmanghi** [cyanogilvie/unix_sockets#3]
the unix_modules repository exploits the git's ability of including as submodules files taken from other repositories, which thusly can be maintained independently. Nonetheless Github tarballs downloaded from the repository tags list are devoid of such files. This is a problem for example attempting to package software for a Linux distribution such as Debian. In fact the missing files cannot be simply added to the source tree because the consistency checks on the source code carried out by the packaging  script would fail. The thread https://stackoverflow.com/questions/34719785/how-to-add-submodule-files-to-a-github-release mentions this fact and suggests a script for creating artifacts to be released that include also submodules files

**Comment by cyanogilvie** [comment]
Yes, I deal with this issue using github workflows to create proper releases on some of my newer projects, like https://github.com/cyanogilvie/tcl-s2n.  Unfortunately urgent commitments for my day job mean I won't be able to set this up for this project before this weekend

**Comment by cyanogilvie** [comment]
A possible alternative that I use in various places is to do a shallow recursive clone rather than fetching a tar of a tag:

```
git clone -b v0.3 --recurse-submodules --shallow-submodules --single-branch --depth 1 https://github.com/cyanogilvie/unix_sockets
```

**Comment by mxmanghi** [comment]
Not a problem: the world out there can safely hold on and wait for the 
next weekend.

**Comment by cyanogilvie** [comment]
I've hacked together the release action, and released v0.5 to test it.  Let me know if this works for you

**Comment by mxmanghi** [comment]
Hi, the 0.5 release package seems to everything needed to build painlessly. Many thanks for your help
